src/Workspaces/Core/Portable/EmbeddedLanguages/LanguageServices/AbstractEmbeddedLanguageProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices
+{
+    /// <summary>
+    /// Abstract implementation of the C# and VB embedded language providers.
+    /// </summary>
+    internal abstract class AbstractEmbeddedLanguageProvider : IEmbeddedLanguageProvider
+    {
+        private readonly ImmutableArray<IEmbeddedLanguage> _embeddedLanguages;
+
+        protected AbstractEmbeddedLanguageProvider(
+            int stringLiteralKind,
+            ISyntaxFactsService syntaxFacts,
+            ISemanticFactsService semanticFacts,
+            IVirtualCharService virtualCharService)
+        {
+            _embeddedLanguages = ImmutableArray.Create<IEmbeddedLanguage>();
```

Comment:
Note: this will be the place in the Regex/Json PRs where we'll add the JsonEmbeddedLanguage and RegexEmbeddedLanguage so that all these features light up with these as the backing systems to handle things.

Reply:
Yeahm this looks weird to have none of these imputs ever used but I get the reason why.  Can we add a comment here explaining why this constructior looks the way iit does?

Reply:
Done!